File: FE/WEB/app/models/geo-subregion.js
import Model, { attr, belongsTo, hasMany } from '@ember-data/model';

export default class GeoSubregionModel extends Model {
  @belongsTo('geo-region', { async: true, inverse: 'geoSubregions' }) geoRegion;
  @hasMany('geo-country', { async: true, inverse: 'geoSubregion' })
  GeoCountries;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) translations;
  @attr('number', { defaultValue: 0 }) geoRegionId;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/GeoSubregion.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using MIT.Fwk.Core.Attributes;
using System.Collections.Generic;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [SkipJwtAuthentication]
    [SkipClaimsValidation]
    [SkipRequestLogging]
    public class GeoSubregion : Identifiable<int>
    {
        [Attr]
        public string Name { get; set; }
        [Attr]
        public string Translations { get; set; }
        [Attr]
        public int GeoRegionId { get; set; }

        [HasOne]
        public virtual GeoRegion GeoRegion { get; set; }
        [HasMany]
        public virtual ICollection<GeoCountry> GeoCountries { get; set; }

    }
}
